classes/requests/helpers/class-ledyer-order.php

```diff
@@ -94,7 +94,7 @@ private function process_order_lines() {
 
 			// Add order line similar to cart data (without explicit rounding).
 			$this->order_lines[] = array(
-				'reference'      => $product->get_sku() ? $product->get_sku() : $product->get_id(),
+				'reference'      => $product->get_sku() ? $product->get_sku() : '',
```

Comment:
I don't think this is the proper solution in this case. Was the issue that there was no product? Or why did we remove the fallback for the product id?
I would suggest that we use the SKU or Product id, but if we don't have either of them, possibly use the `$item_id` from the loop instead? 

Also i see the cart has this added to the reference: `substr( (string) $item_reference, 0, 64 );` to limit it to 64 characters, since im assuming thats a limit by Ledyer. We should probably add that to the shipping and fee as well just to be safe. As well as the fallback to use the order item id in cases where the reference normally is empty.

Reply:
Ledyer does not seem to allow the product id set as reference, it does however allow an empty string. Martin noticed this yesterday, as his InstaWP environment did not have SKU set for its products. Perhaps $item_id would be allowed?

I'll make sure to add the limit!